In [14]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt
import seaborn as sns
%matplotlib inline

# For this homework is used housing dataset

In [3]:
df=pd.read_csv('./data/2023_housing.csv')

In [4]:
df

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
0        -122.23     37.88                41.0        880.0           129.0   
1        -122.22     37.86                21.0       7099.0          1106.0   
2        -122.24     37.85                52.0       1467.0           190.0   
3        -122.25     37.85                52.0       1274.0           235.0   
4        -122.25     37.85                52.0       1627.0           280.0   
...          ...       ...                 ...          ...             ...   
20635    -121.09     39.48                25.0       1665.0           374.0   
20636    -121.21     39.49                18.0        697.0           150.0   
20637    -121.22     39.43                17.0       2254.0           485.0   
20638    -121.32     39.43                18.0       1860.0           409.0   
20639    -121.24     39.37                16.0       2785.0           616.0   

       population  households  median_income  median_house_value  \
0           322.0       126.0         8.3252            452600.0   
1          2401.0      1138.0         8.3014            358500.0   
2           496.0       177.0         7.2574            352100.0   
3           558.0       219.0         5.6431            341300.0   
4           565.0       259.0         3.8462            342200.0   
...           ...         ...            ...                 ...   
20635       845.0       330.0         1.5603             78100.0   
20636       356.0       114.0         2.5568             77100.0   
20637      1007.0       433.0         1.7000             92300.0   
20638       741.0       349.0         1.8672             84700.0   
20639      1387.0       530.0         2.3886             89400.0   

      ocean_proximity  
0            NEAR BAY  
1            NEAR BAY  
2            NEAR BAY  
3            NEAR BAY  
4            NEAR BAY  
...               ...  
20635          INLAND  
20636          INLAND  
20637          INLAND  
20638          INLAND  
20639          INLAND  

[20640 rows x 10 columns]

In [18]:
df.dtypes

longitude             float64
latitude              float64
housing_median_age    float64
total_rooms           float64
total_bedrooms        float64
population            float64
households            float64
median_income         float64
median_house_value    float64
ocean_proximity        object
dtype: object

# Question 1

What's the version of Pandas that you installed?

In [20]:
pd.__version__

'2.3.1'

# Question 2

How many columns are in the dataset?

- 10
- 6560
- 10989
- 20640

In [5]:
len(df)

20640

# Question 3
 Which columns in the dataset have missing values?

- total_rooms
- total_bedrooms
- both of the above
- no empty columns in the dataset

# Question 4

How many unique values does the ocean_proximity column have?

- 3
- 5
- 7
- 9

In [9]:
len(df.ocean_proximity.unique())

5

# Question 5

What's the average value of the median_house_value for the houses located near the bay?

- 49433
- 124805
- 259212
- 380440

In [25]:
df[df.ocean_proximity == 'NEAR BAY'].median_house_value.mean()

np.float64(259212.31179039303)

# Question 6

Calculate the average of total_bedrooms column in the dataset.
Use the fillna method to fill the missing values in total_bedrooms with the mean value from the previous step.
Now, calculate the average of total_bedrooms again.
Has it changed?
Has it changed?

Hint: take into account only 3 digits after the decimal point.

- Yes
- No

# Question 7
Select all the options located on islands.
Select only columns housing_median_age, total_rooms, total_bedrooms.
Get the underlying NumPy array. Let's call it X.
Compute matrix-matrix multiplication between the transpose of X and X. To get the transpose, use X.T. Let's call the result XTX.
Compute the inverse of XTX.
Create an array y with values [950, 1300, 800, 1000, 1300].
Multiply the inverse of XTX with the transpose of X, and then multiply the result by y. Call the result w.
What's the value of the last element of w?
Note: You just implemented linear regression. We'll talk about it in the next lesson.

- -1.4812
- 0.001
- 5.6992
- 23.1233